### PR TITLE
remove direct SQL requests

### PR DESCRIPTION
**Pull request by btry**
code cleanup  task

### CODE REVIEW

File: inc/targetticket.class.php [hunk 1]
```diff
@@ -1088,16 +1088,15 @@ public function save(PluginFormcreatorFormanswer $formanswer)
...
+      $targetTicketId = $this->getID();
       $solo_requester = false;
-      $query_requester = "SELECT id, actor_type, actor_value, use_notification
-                          FROM glpi_plugin_formcreator_targettickets_actors
-                          WHERE plugin_formcreator_targettickets_id = ".$this->getID()."
-                          AND actor_role = 'requester'";
-      $result_requester = $DB->query($query_requester);
+      $targetTicketActor = new PluginFormcreatorTargetTicket_Actor();
+      $rows = $targetTicketActor->find("`plugin_formcreator_targettickets_id` = $targetTicketId
+            AND `actor_role` = 'requester'");
```
Comment (orthagh) [inline inc/targetticket.class.php]: CS: align with "` above

File: inc/targetticket.class.php [hunk 2]
```diff
@@ -1243,8 +1242,8 @@ public function save(PluginFormcreatorFormanswer $formanswer)
       }
 
       // Define urgency
-      $found  = $answer->find('plugin_formcreator_formanwers_id = '.$formanswer->fields['id'].
-            ' AND plugin_formcreator_question_id = '.$this->fields['urgency_question']);
+      $found  = $answer->find('`plugin_formcreator_formanwers_id` = '.$formanswer->fields['id'].
+            ' AND `plugin_formcreator_question_id` = '.$this->fields['urgency_question']);
```
Comment (orthagh) [inline inc/targetticket.class.php]: CS also

File: inc/targetticket.class.php [hunk 3]
```diff
@@ -1318,11 +1317,9 @@ public function save(PluginFormcreatorFormanswer $formanswer)
       ));
 
       // Add actors to ticket
-      $query = "SELECT id, actor_role, actor_type, actor_value, use_notification
-                FROM glpi_plugin_formcreator_targettickets_actors
-                WHERE plugin_formcreator_targettickets_id = " . $this->getID();
-      $result = $DB->query($query);
-      while ($actor = $DB->fetch_array($result)) {
+      $targetTicketActor = new PluginFormcreatorTargetTicket_Actor();
+      $rows = $targetTicketActor->find("`plugin_formcreator_targettickets_id` = '$targetTicketId'");
```
Comment (orthagh) [inline inc/targetticket.class.php]: Here, you change from $this->getID() to targetTicketId.
Wanted change ?

Comment (btry) [inline inc/targetticket.class.php]: yes this is wanted because I dislike strinig concatenation : it makes SQL less readable in my point of view.

Comment (orthagh) [inline inc/targetticket.class.php]: fine for me, i missed the new 1091 line

**Comment by orthagh** [comment]
Ok for me, could we merge, or you wanted to add additional stuff ?

**Comment by btry** [comment]
I wish to merge : I need them to continue with the new actor field. If not we will have to solve a conflict later

**Pull request merged**